otcextensions/sdk/cloud/dds.py

```diff
@@ -0,0 +1,190 @@
+
+        :returns: The results of instance creation
+        :rtype: :class:`~otcextensions.sdk.dds.v3.instance.Instance`
+        """
+
+        datastore_type = kwargs.get('datastore_type')
+        datastore_version = kwargs.get('datastore_version')
+        datastore_storage_engine = kwargs.get('datastore_storage_engine')
+        region = kwargs.get('region')
+        availability_zone = kwargs.get('availability_zone')
+        router = kwargs.get('router')
+        network = kwargs.get('network')
+        security_group = kwargs.get('security_group')
+        password = kwargs.get('password')
+        disk_encryption_id = kwargs.get('disk_encryption_id')
+        mode = kwargs.get('mode')
+        flavors = kwargs.get('flavors')
+        backup_timeframe = kwargs.get('backup_timeframe')
+        backup_keepdays = kwargs.get('backup_keepdays')
+        ssl_option = kwargs.get('ssl_option')
+
+        attrs = {}
+        attrs['name'] = name
+
+        if datastore_type:
+            datastore = {
+                'type': datastore_type,
+                'version': datastore_version,
+                'storage_engine': datastore_storage_engine
+            }
+            attrs['datastore'] = datastore
+        elif datastore_type or datastore_version or datastore_storage_engine:
+            raise exceptions.SDKException(
+                '`datastore_type` and `datastore_version`'
+                ' and `datastore_storage_engine` must be passed together'
+            )
+
+        if region:
+            if region not in ['eu-de', 'eu-nl']:
+                raise exceptions.SDKException(
+                    '`eu-de` or `eu-nl` are supported values'
+                )
+            attrs['region'] = region
+
+        if availability_zone:
+            attrs['availability_zone'] = availability_zone
+
+        if router:
+            router_obj = self.network.find_router(router, ignore_missing=False)
+            attrs['vpc_id'] = router_obj.id
+
+        if network:
+            network_obj = self.network.find_network(
+                network, ignore_missing=False)
+            attrs['subnet_id'] = network_obj.id
+
+        if security_group:
+            security_group_obj = self.network.find_security_group(
+                security_group, ignore_missing=False)
+            attrs['security_group_id'] = security_group_obj.id
+
+        if password:
+            attrs['password'] = password
+
+        if disk_encryption_id:
+            attrs['disk_encryption_id'] = disk_encryption_id
+
+        if mode:
+            if mode not in ['Sharding', 'ReplicaSet']:
+                raise exceptions.SDKException(
+                    '`Sharding` or `ReplicaSet` are supported values'
+                )
+            attrs['mode'] = mode
+
+        flavors_ref = list(self.dds.flavors(
+            region=region,
+            engine_name=datastore_type)
+        )
+        flavors_specs = [flavor.spec_code for flavor in flavors_ref]
+        if flavors:
```

Comment:
There's no need in `if flavors` with following `for flavor in flavors` ― cycle won't be entered if list is empty

Reply:
Since `flavor` is also mandatory, perhaps it is possible to have `flavors = kwargs.get('flavors', [])` and avoid unnecessary ìf`?